### PR TITLE
Feature/931 alfresco model packaging

### PR DESCRIPTION
**Pull request by timijntema**
The only part in the pipeline that does not work at the moment is the docker  push. This requires privileges on the Openzaak dockerhub and I don't have those privileges.

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@aboerma can you please give this a review? :-)

File: openzaak-alfresco-platform-docker/pom.xml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>openzaak-alfresco-platform-docker</artifactId>
+    <name>Alfresco Platform/Repository Docker Module</name>
+    <description>Platform/Repo Docker Module to generate the final Docker image</description>
+    <packaging>jar</packaging>
+
+    <parent>
+        <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline openzaak-alfresco-platform-docker/pom.xml]: can we change this into `org.openzaak`?

File: openzaak-alfresco-platform-docker/pom.xml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        <artifactId>openzaak-alfresco</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <properties>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline openzaak-alfresco-platform-docker/pom.xml]: Same here - `org.openzaak` is vendor neutral :)

File: openzaak-alfresco-platform-docker/pom.xml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            <artifactId>openzaak-alfresco-platform</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline openzaak-alfresco-platform-docker/pom.xml]: another one :)

File: openzaak-alfresco-platform/pom.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>openzaak-alfresco-platform</artifactId>
+    <name>Alfresco Platform/Repository JAR Module</name>
+    <description>Platform/Repo JAR Module (to be included in the alfresco.war) - part of AIO - SDK 4.0
+    </description>
+    <packaging>jar</packaging>
+
+    <parent>
+        <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline openzaak-alfresco-platform/pom.xml]: see earlier remark

File: openzaak-alfresco-share/pom.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>openzaak-alfresco-share</artifactId>
+    <name>Alfresco Share JAR Module</name>
+    <packaging>jar</packaging>
+    <description>Sample Share JAR Module (to be included in the share.war) - part of AIO - SDK 4.0</description>
+
+    <parent>
+        <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/pom.xml]: see earlier remark

File: openzaak-alfresco-share/src/main/resources/META-INF/share-config-custom.xml [hunk 6]
```diff
@@ -0,0 +1,466 @@
+<alfresco-config>
+    <author>OPEN.satisfaction namens Openzaak</author>
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/META-INF/share-config-custom.xml]:
```suggestion
    <author>OPEN.satisfaction</author>
```
Seems sufficient to give due credit, no?

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties [hunk 7]
```diff
@@ -0,0 +1,95 @@
+# Type labels
+type.drc_zaaktypefolder=Openzaak zaaktype folder
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]: Please stylize Open Zaak either as `Open Zaak` or `OpenZaak`

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+type.drc_zaakfolder=Openzaak zaakfolder
+
+type.drc_document=Openzaak document
+type.drc_gebruiksrechten=Openzaak gebruiksrechten
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]: this is rather `Documenten API gebruiksrechten`, this applies for everything of `type.drc_`

Could maybe also be named "Open Zaak Documenten API gebruiksrechten" but that's quite verbose.

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+form.advancedsearch.document.label=Document zoeken
+form.advancedsearch.document.description=Zoeken naar Openzaak documenten
+
+form.document.set.basisinformatie.label=Basis informatie
+form.document.set.basisinformatie.uuid.label=Document UUID
+form.document.set.basisinformatie.bestandsnaam.label=Bestandsnaam
+form.document.set.basisinformatie.identificatie.label=Identificatie
+form.document.set.basisinformatie.titel.label=Titel
+form.document.set.basisinformatie.beschrijving.label=Beschrijving
+form.document.set.basisinformatie.informatieobjecttype.label=Informatie object type
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]:
```suggestion
form.document.set.basisinformatie.informatieobjecttype.label=Informatieobjecttype
```

Could also use "documenttype" which is less abstract for end users.

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+form.document.set.basisinformatie.creatiedatum.label=Creatie datum
+form.document.set.basisinformatie.auteur.label=Auteur
+form.document.set.basisinformatie.taal.label=Taal
+
+form.document.set.documentinformatie.label=Document informatie
+form.document.set.documentinformatie.bronorganisatie.label=Bron organisatie
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]:
```suggestion
form.document.set.documentinformatie.bronorganisatie.label=Bronorganisatie
```

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+form.document.set.integriteitsinformatie.integriteitwaarde.label=Integriteit waarde
+form.document.set.integriteitsinformatie.integriteitdatum.label=Integriteitdatum
+form.document.set.integriteitsinformatie.integriteitalgoritme.label=Integriteitalgoritme
+
+form.document.set.gebruiksinformatie.label=Gebruik document
+form.document.set.gebruiksinformatie.vertrouwelijkaanduiding.label=Vertrouwelijkheids aanduiding
+form.document.set.gebruiksinformatie.indicatiegebruiksrecht.label=Indicatie gebruiksrecht
+
+# OIO messages
+form.advancedsearch.oio.label=OIO zoeken
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]:
```suggestion
form.advancedsearch.oio.label=ObjectInformatieObject zoeken
```
& volgende entries kunnen dan ook volledige woord gebruiken

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/openzaak-alfresco-share-slingshot-application-context.xml [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+    Define beans related to this Share extension here, such as
+     evaluators and resource bootstrapping.
+
+     Note. Any configuration related to the whole Share.war webapp
+           (i.e. not specific to this share extension) should be
+            placed in the alfresco-allinone/share/src/main/resources/alfresco/web-extension/custom-slingshot-application-context.xml file
+   -->
+
+   <!-- Add module specific messages and labels -->
+   <bean id="nl.opensatisfaction.openzaak-alfresco-share.resources"
```
Comment (sergei-maertens) [inline openzaak-alfresco-share/src/main/resources/alfresco/web-extension/openzaak-alfresco-share-slingshot-application-context.xml]:
```suggestion
   <bean id="org.openzaak.openzaak-alfresco-share.resources"
```

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,439 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>nl.opensatisfaction</groupId>
```
Comment (sergei-maertens) [inline pom.xml]:
```suggestion
    <groupId>org.openzaak</groupId>
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Some high-level remarks - mostly about really presenting this as something belonging to the Open Zaak community rather than implied ownership by Open.satisfaction. Open Zaak as a community project should stay neutral. Couldn't find a hard requirement, but the closest thing is probably https://github.com/open-zaak/open-zaak/blob/main/docs/introduction/open-source/public_code.md#create-reusable-and-portable-code.

Content-wise this looks fine to me with my very limited knowledge of Java stacks :-)

**Comment by aboerma** [comment]
Heb bovenstaande feature lokaal uitgechecked, gebuild en getest. Voor zover ik kan zien werkt alles zoals is beschreven in de README.md. Inhoudelijk heb ik geen opmerkingen, works as advertised.

File: openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties (hunk 8 above)
Comment (timijntema) [reply openzaak-alfresco-share/src/main/resources/alfresco/web-extension/messages/openzaak-alfresco-share.properties]: I will be using the shorter version. The window those labels are shown in is not that large

**Comment by timijntema** [comment]
@sergei-maertens I made the requested changes. Please check the PR again.

**Comment by sergei-maertens** [comment]
@timijntema thanks, will get to this ASAP, should be done some time this week

**Review by sergei-maertens — APPROVED** [review]
Thanks, let's get this merged :pray:

**Pull request merged**